docs/security-privacy-self-review.md

```diff
@@ -0,0 +1,85 @@
+
+Implementors will want to carefully consider what information is revealed in the error message when `share()` is rejected. Even distinguishing between the case where no targets are available and user cancellation could reveal information about which apps are installed on the user's device. The previous version of the specification distinguished when sharing failed via `DataError` instead, which may have exposed unnecessary information.
+
+## 2.3 How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+
+Any information shared was already exposed to the web site; no further personal information is revealed by the API to the web site.
+
+The receiving share target selected by the user can be given arbitrary information by the web site via the `ShareData`'s properties. In most expected user interfaces, the user is not made aware of what data is being shared by the web site with the target. This could lead to a web site in posession of personal information sharing it with a target without the user's knowledge. One mitigation would be to have the user interface allow inspection of the data.
+
+## 2.4 How does this specification deal with sensitive information?
+
+As in the previous question, the specification does not limit what data is shared, and the data could include sensitive information already revealed to the web site. However, the user is in control of which, if any, target application will receive the information.
+
+## 2.5 Does this specification introduce new state for an origin that persists across browsing sessions?
+
+No.
+
+## 2.6 What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+The presence of the API may reveal characteristics of the user's device beyond the user agent. For example, the user agent may only expose the `navigator.share()` API on "mobile" devices, revealing the type of device the user agent is executing on.
+
+Revealing content types supported via `files` should be done carefully, as `canShare()` does not require
+[transient activation](https://html.spec.whatwg.org/multipage/interaction.html#transient-activation). For example, if the `canShare()` API response was conditional on what native applications were installed on the device, this could be used to fingerprint the user.
+
+## 2.7 Does this specification allow an origin access to sensors on a user’s device
+
+No.
+
+## 2.8 What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+The data exposed to an origin is the presence of the API itself, whether certain content types are supported, and the results of interacting with the API (error or success).
```

Comment:
Again, #108 might mitigate "whether certain content types are supported" to a degree. It would be good reinforce that a site can't distinguish between the user cancelled and no share target supports that content type.

Reply:
Thanks - updated!